src/main/java/org/spongepowered/api/net/PlayerNetworkHandler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.net;
+
+import org.spongepowered.api.entity.player.Player;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+
+/**
+ * Represents a connection of a client to the server.
+ */
+public interface PlayerNetworkHandler {
+
+    /**
+     * Gets the associated Player that this player connection is associated
+     * with.
+     *
+     * @return The associated player
+     */
+    Player getPlayer();
+
+    /**
+     * Gets the {@link InetAddress} of this connection.
+     *
+     * @return The address
+     */
+    InetAddress getAddress();
```

Comment:
Is there any special reason to have this as `InetAddress` instead of `InetSocketAddress` which also provides the client port? Also this conflicts with the addition of `getAddress()` in #316 so we need to removed it in one of the pull request (better there, because this pull request only contains things like that).